### ISSUE TITLE
Animator scale off in developer settings no longer calls end listener (snapshot)

### ISSUE THREAD

**Issue by kakai248** [airbnb/lottie-android#1386]
We have a screen where we depend on having a certain lottie animation to end. This means we have a `AnimatorListener` and wait for the `onAnimationEnd`.

In 3.1.0-SNAPSHOT, because of #1349, if we have animator scale off in developer settings, the animation stops and the listener never gets called.

Is this something that lottie should fix (by calling the listener anyway) or should we address this in the app? What would make the most sense?

**Comment by gpeal** [comment]
@kakai248 Can you use the issue-repro module in this repo to reproduce this issue? I just tried it and it works for me.

**Comment by kakai248** [comment]
I've managed to fix it, the issue was on our side. But it's a bit weird.

I have some utils class to handle setting the lotties on the view and handling loops and stuff like that. Part of that code was:

```
setAnimation(rawResId)

if (autoPlay) {
    playAnimation()
}
```

This was working fine for every lottie. But with the animator scale off, it stopped working. Now I know setAnimation is async, but it was working fine and I had no problems whatsoever.

I managed to fix the problem by changing the previous code to:

```
view.addLottieOnCompositionLoadedListener {
    if (autoPlay) {
        playAnimation()
    }
}

setAnimation(rawResId)
```

Any idea why the setting could be affecting this?
